noxfile.py

```diff
@@ -60,32 +59,36 @@ def check_manifest(session: nox.Session) -> None:
     session.run("check-manifest", *session.posargs)
 
 
-@nox.session(python=PYTHON_ALL_VERSIONS)
+@nox.session
 def update_constraints(session: nox.Session) -> None:
     """
     Update the dependencies inplace.
     """
-    session.install("pip-tools")
-    assert isinstance(session.python, str)
-    python_version = session.python.replace(".", "")
-    env = os.environ.copy()
-    # CUSTOM_COMPILE_COMMAND is a pip-compile option that tells users how to
-    # regenerate the constraints files
-    env["CUSTOM_COMPILE_COMMAND"] = f"nox -s {session.name}"
-    session.run(
-        "pip-compile",
-        "--allow-unsafe",
-        "--upgrade",
-        "cibuildwheel/resources/constraints.in",
-        f"--output-file=cibuildwheel/resources/constraints-python{python_version}.txt",
-        env=env,
-    )
-    if session.python == PYTHON_ALL_VERSIONS[-1]:
-        RESOURCES = DIR / "cibuildwheel" / "resources"
-        shutil.copyfile(
-            RESOURCES / f"constraints-python{python_version}.txt",
-            RESOURCES / "constraints.txt",
+
+    if getattr(session.virtualenv, "venv_backend", "") != "uv":
+        session.install("uv")
+
+    for minor_version in range(7, 13):
+        python_version = f"3.{minor_version}"
+        env = os.environ.copy()
+        # CUSTOM_COMPILE_COMMAND is a pip-compile option that tells users how to
+        # regenerate the constraints files
+        env["CUSTOM_COMPILE_COMMAND"] = f"nox -s {session.name}"
```

Comment:
Not critical, but it looks like this isn't having an effect on uv yet, found an issue here- https://github.com/astral-sh/uv/issues/1527

Reply:
It's added now. Though IMO it's not really required, as unlike before, you don't need any special setup to compile it.

Reply:
It's called UV_CUSTOM_PIP_COMPILE, even though PR and release notes get it wrong. https://github.com/astral-sh/uv/pull/2554